contracts/NFTVault.sol

```diff
@@ -179,8 +193,8 @@ contract NFTVault is IERC721Receiver, IERC1155Receiver, AccessControl {
         uint256 _tokenId,
         bytes memory
     ) public virtual override returns (bytes4) {
-        if (keys.transfer(_from, TOKEN_AMOUNT) && whiteList[msg.sender]) {
-            nfts.push(NFT("ERC721", msg.sender, _tokenId, 0, 0));
+        if (keys.transfer(_from, TOKEN_AMOUNT) && whiteList[_from]) {
+            nfts.push(NFT(msg.sender, _tokenId, 0));
```

Comment:
`msg.sender` probably would be better to replace it with `_operator` param, as it represents contract address of transfered NFT

Reply:
removed `_operator` param, it was giving the same wallet address that sent the NFT not the ERC721 contract address like `msg.sender`.